### PR TITLE
Add proxy_list and proxy_list_external magic xattrs

### PR DESCRIPTION
**Pull request by mharvey-jt**

### CODE REVIEW

File: cvmfs/magic_xattr.cc [hunk 1]
```diff
@@ -480,6 +482,31 @@ std::string ProxyMagicXattr::GetValue() {
   }
 }
 
+static std::string ListProxy(download::DownloadManager *dm) {
+  vector< vector<download::DownloadManager::ProxyInfo> > proxy_chain;
+  unsigned current_group;
+  dm->GetProxyInfo(
```
Comment (jblomer) [inline cvmfs/magic_xattr.cc]: Doesn't need a line break

File: cvmfs/magic_xattr.cc [hunk 2]
```diff
@@ -480,6 +482,31 @@ std::string ProxyMagicXattr::GetValue() {
...
+    &proxy_chain, &current_group, NULL);
+  std::string buf = "";
+  for (std::vector<vector<download::DownloadManager::ProxyInfo>>::iterator
```
Comment (jblomer) [inline cvmfs/magic_xattr.cc]: That's fine but wouldn't it be easier to use an integer and iterate from zero to `vector::size()`?

**Review by jblomer — CHANGES_REQUESTED** [review]
LGTM, just some nits and let's amend integration test case 087 with the new attributes.

**Comment by mharvey-jt** [comment]
@cernvm-bot unittest

**Comment by mharvey-jt** [comment]
@cernvm-bot unittest linux

File: cvmfs/magic_xattr.cc (hunk 2 above)
Comment (mharvey-jt) [reply cvmfs/magic_xattr.cc]: simplified

File: cvmfs/magic_xattr.cc (hunk 1 above)
Comment (mharvey-jt) [reply cvmfs/magic_xattr.cc]: fixed

**Comment by mharvey-jt** [comment]
added simple check in 087

**Comment by mharvey-jt** [comment]
@cernvm-bot codespell

**Comment by jblomer** [comment]
@cernvm-bot cloudtest

**Review by jblomer — APPROVED** [review]

**Comment by mharvey-jt** [comment]
@cernvm-bot cloudtest

**Comment by mharvey-jt** [comment]
@cernvm-bot cloudtest

**Comment by jblomer** [comment]
@cernvm-bot cloudtest

**Comment by mharvey-jt** [comment]
@cernvm-bot cloudtest

**Comment by mharvey-jt** [comment]
@cernvm-bot cloudtest all

**Review by vvolkl — APPROVED** [review]

**Pull request merged**